File: java/GateFour.java/Level5.java
public class Level5{
    public boolean isPrime(int number) {
        if(number < 2)
            return false;
        for(int divisor = 2; divisor < number; divisor++){
            if (number % divisor == 0){
                return false;
            }
            }

            return true;

    }
    public int [] getPrimeNumber(int [] numbers){
        int size = 0;
            for(int number : numbers){
                if(isPrime(number)) 
                    size++;

        }
          int [] result = new int [size];

            int index = 0;
        for(int number : numbers){
            if(isPrime(number)) 
            result[index++] = number;
            

        }

    

        for (int count = 0; count < result.length; count++){

        for(int checking = count+1; checking < result.length; checking++){
            if (result[count] > result[checking]){
                int temp = result[count];
                result[count] = result[checking];
                result[checking] = temp;

                        }
            }
                
        }
        
        return result;  
        }




        public int [] replacingNegatives(int [] numbers){
            for (int index = 0; index < numbers.length; index++){
                if (numbers[index] < 0)
                    numbers[index] = 0;
         }

            return numbers;

    
        }
          
          public int[] moveZeros (int [] numbers) {
            int [] result = new int [numbers.length];
            int index = 0;
            for(int number: numbers){
                if(number != 0){
                    result[index++] = number;
                }
        }
            return result;
               
                }


        

        }






    

File: python/GateFour/level_5.py
def is_prime(number):
    if number <= 1:
        return False
    for divisor in range(2,number):
        if number % divisor == 0:
            return False
    return True

def get_primes(numbers):
    primes = []
    for number in numbers:
        if is_prime(number):
            primes.append(number)
            primes.sort
    return primes

def replace_negatives(numbers):
    for index in range(len(numbers)):
        if numbers[index] < 0:
            numbers[index] = 0
    return numbers

def move_zeros(numbers):
    result = []
    count_zeros = 0
    for number in numbers:
        if number == 0:
            count_zeros += 1
        else:
            result.append(number)
    result.extend([0] *count_zeros)
    return result

    





def find_duplicates(numbers):
    duplicates = []
    for index in range(len(numbers)):
        for checking in range(index + 1, len(numbers)) :
            if numbers[index] == numbers[checking]:
                if numbers[index ] not in duplicates :
                    duplicates.append(numbers[index])
                    break
    return duplicates
    
